### DISCUSSION TITLE
Issues in iOS SDK

### DISCUSSION THREAD

**Discussion by omerfarukose** [ant-media/Ant-Media-Server#5791] [Q&A]
Hi, 

We are facing some problems in the iOS SDK.

First, in the multitrack conference application, if the first participant has joined using iOS SDK, the first participant cannot use the data channel properly. 

The second issue when we finish the first conference and join the second conference, we have problems with the sound. I think it might be related to the WebScoket disconnection. The WebScoket disconnected listener isn't triggered after the disconnect request, so the functions that should work in the listener cannot be run properly.

Is there any way to overcome these problems ?

Thanks.

**Comment by mekya** [comment]
Hi @omerfarukose,

I just find some opportunity to test this case. Thank you for your patience. 

It seems that there are two issues. 
>First, in the multitrack conference application, if the first participant has joined using iOS SDK, the first participant cannot use the data channel properly.

My colleague(@lastpeony ) fixed the problem and it's under quality check. I hope you understand the integrity of the project is essential for us so that several quality checkes and reviews should be done before merging into the master. I mean  It's not merged into the master yet. I hope we can merge it in a week. Fortunately, my colleagues can give snapshot to test it on your end. 

>The second issue when we finish the first conference and join the second conference, we have problems with the sound...

I wrote a simple report to the issue below. I think it's the same issue.  
https://github.com/ant-media/Ant-Media-Server/issues/5808

Please let me know what the differences are between your scenario and my scenario as I mention in issue.

**Reply by mekya** [reply]
Hi @omerfarukose 
>First, in the multitrack conference application, if the first participant has joined using iOS SDK, the first participant cannot use the data channel properly.

I've also merged the data channel issue -> https://github.com/ant-media/Ant-Media-Server/issues/5667
The issue is reproducible in JS SDK. I'll also try to confirm the fix with iOS SDK.

>The second issue when we finish the first conference and join the second conference, we have problems with the sound...
This one is also reproduced on my end by changing the reproduce scenario. It's fixed and merged

The snapshot that includes both fixes has passed the CI/CD. It's ready to test it on your end. 

FYI
